### ISSUE TITLE
What-If should understand resources that are managed by other resources declared in the template

### ISSUE THREAD

**Issue by alex-frankel** [Azure/arm-template-whatif#78]
Examples:
* VMs will create disks
* AKS will setup a VMSS cluster
* DeploymentScripts will create ACI + Storage

**Comment by alex-frankel** [comment]
Customer reported examples:
* https://github.com/Azure/arm-template-whatif/issues/67

**Comment by JustinGrote** [comment]
Here's an example I have. I'm attaching VM domain join extensions via a looped reference to another nested deployment. These don't show up at all in -whatif. If whatif can't evaluate these it should at least show a warning that it expects additional resources of a certain type to be created.

![image](https://user-images.githubusercontent.com/15258962/83335666-2ab21f80-a263-11ea-8c6d-69d95ff3879c.png)

**Comment by alex-frankel** [comment]
@JustinGrote I think this is actually in issue with the reference() call. What-if is able to evaluate nested templates, but if there are references() to resolve the name or resourceId of the resources in the nested template, what-if will not find those resources since it doesn't know what to look for.

We have a separate issue (#83) tracking cases related to the reference() function. Can you add this there?

**Comment by JustinGrote** [comment]
@alex-frankel it may help in the output to note it as "calculated" or maybe have a new whatif status of "Inconclusive" to indicate that ARM doesn't know if this will change or not because it can't evaluate the reference. This is how Terraform handles states like this. It *should* evalute the reference or at least guess if possible (my example it would be guessable), but I understand that would probably be a large engineering effort with a lot of edge cases.

**Comment by alex-frankel** [comment]
Yeah, I agree we should provide more information as to what as happening. We will discuss this during our triage this week. cc @shenglol @majastrz 

If we implement #83, it should resolve the issue you are hitting

**Comment by shenglol** [comment]
Another customer reported example for NSG: https://github.com/Azure/arm-template-whatif/issues/60.

**Comment by shenglol** [comment]
https://github.com/Azure/arm-template-whatif/issues/46.
